sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -100,10 +100,37 @@ inline __SYCL_ALWAYS_INLINE
 #endif
 }
 
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE std::enable_if_t<std::is_same<T, half>::value ||
+                                                 std::is_same<T, float>::value,
+                                             sycl::marray<T, N>>
+tanh(sycl::marray<T, N> x) __NOEXC {
+  sycl::marray<T, N> res;
+#if defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
+  for (size_t i = 0; i < N / 2; i++) {
+    auto partial_res = __clc_native_tanh(sycl::detail::to_vec2(x, i * 2));
```

Comment:
`native::tanh` above does some conversion to OCL types before calling `__clc_native_tanh`. Should this also be done here? Could this just call `native::tanh` and rely on the vector implementation of that?

Reply:
So I tested this with the CUDA and OpenCL backends and found no difference whether we use these conversion functions or not: from what I read there isn't a reason to cast from float to cl_float if you are only dealing with device code although I may have misunderstood and there may be some devices where this is necessary; for consistency I've used the OCL conversion stuff via calling the native::tanh vector implementations as you suggest.